client/src/actions/team.js

```diff
@@ -1,15 +1,70 @@
 /**
  * This file houses our team-related Action Creators.
  */
-import * as api from '../api';
+import * as api from "../api";
 import {
-    LINK_TEAM_TWITTER, 
-    UNLINK_TEAM_TWITTER, 
-    FETCH_TEAM_INFO,     
-    GET_TEAM_MEMBERS_BY_TEAM_ID, 
-    CREATE_TEAM_MEMBER, 
-    UPDATE_TEAM_MEMBER, 
-    DELETE_TEAM_MEMBER } from './types';
+  LINK_TEAM_TWITTER,
+  UNLINK_TEAM_TWITTER,
+  FETCH_TEAM_INFO,
+  GET_TEAM_MEMBERS_BY_TEAM_ID,
+  CREATE_TEAM_MEMBER,
+  UPDATE_TEAM_MEMBER,
+  DELETE_TEAM_MEMBER,
+  ADD_TEAM,
+} from "./types";
+
+/**
+ * Adds a new team to redux store and database. If api request fails the team state's requestFailed attribute will be 'true'
+ * @param teamInfo contains teamName, orgName and emailnot
+ */
+export const addTeamInfo = (teamInfo, result) => async (dispatch) => {
+  try {
+    const data = await api.addTeam(teamInfo);
+    const teamData = {
+      ...teamInfo,
+      teamId: data.data._id,
+      requestFailed: false,
```

Comment:
I don't really get this part... just wondering why is it hard-coded like this (?)

Reply:
To add, if the api request fails, requestFailed will be true, but in the catch I don't see it being set the true?

Reply:
my bad, that variable should not be there, I forgot to remove it